src/ConstructiveSolidGeometry/SurfacePrimitives/EllipsoidMantle.jl

```diff
@@ -0,0 +1,168 @@
+@with_kw struct EllipsoidMantle{T,RT,TP,TT,D} <: AbstractCurvedSurfacePrimitive{T}
+    r::RT = 1
+    φ::TP = nothing
+    θ::TT = nothing
+
+    origin::CartesianPoint{T} = zero(CartesianPoint{T})
+    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+end
+
+flip(em::EllipsoidMantle{T,RT,TP,TT,:inwards}) where {T,RT,TP,TT} = 
+    EllipsoidMantle{T,RT,TP,TT,:outwards}(em.r, em.φ, em.θ, em.origin, em.rotation )
+
+const FullSphereMantle{T,D} = EllipsoidMantle{T,T,Nothing,Nothing,D}
+const FullEllipsoidMantle{T,D} = EllipsoidMantle{T,NTuple{3,T},Nothing,Nothing,D}
+
+function lines(em::FullSphereMantle{T}) where {T} 
+    ellipse_xy = Ellipse{T,T,Nothing}(r = em.r[1], φ = em.φ, origin = em.origin, rotation = em.rotation)
+    ellipse_xz = Ellipse{T,T,Nothing}(r = em.r[1], φ = em.φ, origin = em.origin, rotation = em.rotation * RotX(T(π)/2))
+    ellipse_yz = Ellipse{T,T,Nothing}(r = em.r[1], φ = em.φ, origin = em.origin, rotation = em.rotation * RotX(T(π)/2) * RotY(T(π)/2))
+    (ellipse_xy, ellipse_xz, ellipse_yz)
+end
+function lines(em::FullEllipsoidMantle{T}) where {T} 
+    ellipse_xy = Ellipse{T,NTuple{2, Tuple{T}},Nothing}(r = ((em.r[1],), (em.r[2],)), φ = em.φ, origin = em.origin, rotation = em.rotation)
+    ellipse_xz = Ellipse{T,NTuple{2, Tuple{T}},Nothing}(r = ((em.r[1],), (em.r[3],)), φ = em.φ, origin = em.origin, rotation = em.rotation * RotX(T(π)/2))
+    ellipse_yz = Ellipse{T,NTuple{2, Tuple{T}},Nothing}(r = ((em.r[2],), (em.r[3],)), φ = em.φ, origin = em.origin, rotation = em.rotation * RotX(T(π)/2) * RotY(T(π)/2))
+    (ellipse_xy, ellipse_xz, ellipse_yz)
+end
+
+extremum(e::EllipsoidMantle{T,T}) where {T} = e.r
+extremum(e::EllipsoidMantle{T,NTuple{3,T}}) where {T} = max(e.r)
```

Comment:
Does this work? I assume it should be `maximum(e.r)`..

Reply:
Correct, it does not work.. I fixed it